gensim/models/_fasttext_bin.py

```diff
@@ -225,19 +227,48 @@ def _load_matrix(fin, new_format=True):
 
     num_vectors, dim = _struct_unpack(fin, '@2q')
 
-    float_size = struct.calcsize('@f')
-    if float_size == 4:
-        dtype = np.dtype(np.float32)
-    elif float_size == 8:
-        dtype = np.dtype(np.float64)
-    else:
-        raise ValueError("Incompatible float size: %r" % float_size)
-
-    matrix = np.fromfile(fin, dtype=dtype, count=num_vectors * dim)
+    count = num_vectors * dim
+    matrix = _fromfile(fin, None, count)
+    assert matrix.shape == (count,), 'expected (%r,),  got %r' % (count, matrix.shape)
     matrix = matrix.reshape((num_vectors, dim))
     return matrix
 
 
+def _batched_generator(fin, count, batch_size=1e6):
+    """Reads count floats from fin.
+
+    Batches up read calls to avoid I/O overhead.  Keeps no more than batch_size
+    floats in memory at once.
+
+    Yields floats.
+
+    """
+    while count > batch_size:
+        batch = _struct_unpack(fin, '@%df' % batch_size)
```

Comment:
`.format` maybe or you prefer old-style formatting?

Reply:
Yeah, my personal preference is old style.

The .format stuff is the latest and greatest for now, but will get replaced by fstrings once Py2 support gets dropped.